index.bs

```diff
@@ -0,0 +1,565 @@
+
+If |this|.{{[[state]]}} is `"started"`, |this|.{{userState}} is {{"active"}} and
+the user has not interacted with the device in at least |this|.{{[[threshold]]}}
+milliseconds, the [=user agent=] MUST [=queue a task=] to run the following
+steps:
+
+1.  Set |this|.{{[[userState]]}} to {{"idle"}}.
+1.  [=Fire an event=] named {{"change"}} at |this|.
+
+<wpt>
+  interceptor.https.html
+</wpt>
+</div>
+
+<div algorithm="user becomes active">
+
+If |this|.{{[[state]]}} is `"started"`, |this|.{{userState}} is {{"idle"}} and
+the user interacts with the device, the [=user agent=] MUST [=queue a task=] to
+run the following steps:
+
+1.  Set |this|.{{[[userState]]}} to {{"active"}}.
+1.  [=Fire an event=] named {{"change"}} at |this|.
+
+<wpt>
+  interceptor.https.html
+</wpt>
+</div>
+
+### {{IdleDetector/screenState}} attribute ### {#api-idledetector-screenstate}
+
+<div algorithm>
+The <dfn attribute for=IdleDetector>screenState</dfn> getter steps are:
+
+1.  Return [=this=].{{[[screenState]]}}.
+
+</div>
+
+<div algorithm="screen locked">
+
+If |this|.{{[[state]]}} is `"started"` and the device is locked by the user, an
+automatic timeout or any other event, the [=user agent=] MUST [=queue a task=] to
+run the following steps:
+
+1.  Set |this|.[[{{screenState}}]] to {{"locked"}}.
+1.  [=Fire an event=] named {{"change"}} at |this|.
+
+<wpt>
+  interceptor.https.html
+</wpt>
+</div>
+
+<div algorithm="screen unlocked">
+
+If |this|.{{[[state]]}} is `"started"` and the device is unlocked by the user or
+for any other reason, the [=user agent=] MUST [=queue a task=] to run the
+following steps:
+
+1.  Set |this|.[[{{screenState}}]] to {{"unlocked"}}.
+1.  [=Fire an event=] named {{"change"}} at |this|.
+
+<wpt>
+  interceptor.https.html
+</wpt>
+</div>
+
+### {{IdleDetector/onchange}} attribute ### {#api-idledetector-onchange}
+
+<dfn attribute for="IdleDetector">onchange</dfn> is an <a>Event handler IDL
+attribute</a> for the {{change}} event type.
+
+### {{IdleDetector/requestPermission()}} method ### {#api-idledetector-requestpermission}
+
+<div algorithm>
+The <dfn method for=IdleDetector>requestPermission()</dfn> method steps are:
+
+1.  Let |result| be [=a new promise=].
+1.  [=Queue a task=] to run these steps:
+    1.  If the [=relevant global object=] of [=this=] does not have [=transient
+        activation=], [=reject=] |result| with a "{{NotAllowedError}}"
+        {{DOMException}} and abort these steps.
+    1.  Let |permissionState| be the result of [=requesting permission to use=]
+        the [=powerful feature=] named `"idle-detection"`.
+    1.  Resolve |result| with |permissionState|.
+1. Return |result|.
+
+</div>
+
+### {{IdleDetector/start()}} method ### {#api-idledetector-start}
+
+<div algorithm>
+
+The <dfn method for=IdleDetector>start(|options|)</dfn> method, when invoked,
+MUST run these steps:
+
+1.  Let |result| be [=a new promise=].
+1.  If the [=relevant global object=]'s associated {{Document}} is not [=allowed
```

Comment:
"associated Document" should link to https://html.spec.whatwg.org/#concept-document-window